ra/ra.go

```diff
@@ -1470,17 +1482,6 @@ func (ra *RegistrationAuthorityImpl) UpdateAuthorization(
 		return core.Authorization{}, berrors.InternalServerError("could not compute expected key authorization value")
 	}
 
-	// NOTE(@cpu): Historically challenge update required the client to send
```

Comment:
I think it's fine to remove this but can you add something to the PR desc about this removal (maybe in the second point?). In theory this currently catches some incredibly broken clients that are sending the wrong key authorization and this prevents an unnecessary validation attempt that they will quite likely fail (stats show this happens maybe ~25 times a day).

Reply:
SGTM. I updated the PR description. ~25 extra validations a day shouldn't hurt anyone :-)